app/src/main/java/com/x8bit/bitwarden/data/vault/manager/VaultLockManagerImpl.kt

```diff
@@ -435,6 +436,12 @@ class VaultLockManagerImpl(
         userId: String,
         isAppRestart: Boolean = false,
     ) {
+        val accounts = authDiskSource.userAccountTokens
+        // Check if the user is already logged out. If this is the case no need to check timeout.
+        if (accounts.find { it.userId == userId }?.isLoggedIn?.not() == true) {
+            return
```

Comment:
Is this just an optimization or is this required?

Reply:
this is required, if you switch back to the soft logged out user without this and they have the "immediate" timeout set it will just keep switching to the next user over and over in a loop.

Reply:
⛏️ It would be helpful to note that undesired behavior in the comment.